hi-ml/src/health_ml/run_ml.py

```diff
@@ -396,26 +387,23 @@ def run(self) -> None:
         self.setup()
         try:
             self.init_training()
+
             if not self.container.run_inference_only:
                 # Backup the environment variables in case we need to run a second training in the unit tests.
                 old_environ = dict(os.environ)
-
-                # do training
+                # Do training
                 with logging_section("Model training"):
                     self.run_training()
-
-                # Kill all processes besides rank 0
+                # Update the checkpoint handler state
+                self.checkpoint_handler.additional_training_done()
+                # Kill all processes besides rank 0 after training is done to start inference on a single device
                 self.after_ddp_cleanup(old_environ)
 
-                # load model checkpoint for custom inference or additional validation step
-                if self.container.has_custom_test_step() or self.container.run_extra_val_epoch:
-                    self.load_model_checkpoint()
+            self.init_inference()
+
+            with logging_section("Model Validation"):
```

Comment:
Lower case for consistency with the other sections?
```suggestion
            with logging_section("Model validation"):
```